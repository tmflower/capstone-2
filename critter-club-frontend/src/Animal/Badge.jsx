export function Badge({ animalName, url }) {
    return (
        <div style={{width:200, height:200, backgroundColor:'white', color:'black', margin:'10px'}}>{animalName}
            <img src={url} alt="critter" style={{ height:150 }}/>
        </div>
    )
}