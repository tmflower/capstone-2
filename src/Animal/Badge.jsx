import { Paper, Typography } from "@mui/material";
import { theme } from "../theme/theme";

export function Badge({ animalName, url }) {
    return (
        
        <Paper 
        elevation={18} 
        sx={{ 
            borderRadius: '50%', 
            height: 200, 
            width: 320, 
            display: 'flex', 
            flexDirection: 'column', 
            alignItems: 'center', 
            justifyContent: 'center',
            margin: 2 }}>
                <Typography 
                    variant="h6" 
                    fontFamily={theme.typography.primary}>{animalName}
                    
                </Typography>
                <img 
                    src={url} 
                    alt={animalName}
                    style={{ maxWidth: 150, maxHeight: 130 }}/>        
        </Paper>
         
    )
}