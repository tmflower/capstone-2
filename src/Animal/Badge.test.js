import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Badge } from './Badge';

// smoke test
test("renders without crashing", () => {
    render(
    <MemoryRouter>
      <Badge />
    </MemoryRouter>);
  });

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Badge />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});