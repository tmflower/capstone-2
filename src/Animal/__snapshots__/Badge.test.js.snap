// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation18 css-12399m2-MuiPaper-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1f1ge3c-MuiTypography-root"
    />
    <img
      style="max-width: 150px; max-height: 130px;"
    />
  </div>
</DocumentFragment>
`;
